### PR TITLE
[VAL-3] Implement static Theaters

### PR DESCRIPTION
**Pull request by anagavrilovic**

### CODE REVIEW

**Comment by anagavrilovic** [comment]
[[VAL-3] Implement static Theaters](https://trello.com/c/6koqYS0e/24-val-3-implement-static-theaters)

File: src/main/java/com/example/videotecha/model/Theater.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.SequenceGenerator;
+
+@Entity
+public class Theater {
+
+    @Id
+    @SequenceGenerator(name = "theaterSequenceGenerator", sequenceName = "theaterSequenceGenerator", initialValue = 1, allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "theaterSequenceGenerator")
+    @Column(name="id", unique=true, nullable=false)
```
Comment (pantelicn) [inline src/main/java/com/example/videotecha/model/Theater.java]: `unique` and `nullable` are redundant when `@Id` is applied

File: src/main/java/com/example/videotecha/service/TheaterService.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.example.videotecha.model.Theater;
+
+import java.util.List;
+
+public interface TheaterService {
+
+    List<Theater> findAll();
+
+    Theater findById(Long id);
```
Comment (pantelicn) [inline src/main/java/com/example/videotecha/service/TheaterService.java:11]: Be consistent with formatting, if you add an empty line at the beginning of the class, add it at the end as well.

File: src/main/java/com/example/videotecha/service/impl/TheaterServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@Service
+public class TheaterServiceImpl implements TheaterService {
+
+    TheaterRepository theaterRepository;
```
Comment (pantelicn) [inline src/main/java/com/example/videotecha/service/impl/TheaterServiceImpl.java]: private final

File: src/main/java/com/example/videotecha/service/impl/TheaterServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Override
+    public List<Theater> findAll() {
+        return theaterRepository.findAll();
+    }
+
+    @Override
+    public Theater findById(Long id) {
+        return theaterRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "There is no theater with this id."));
```
Comment (pantelicn) [inline src/main/java/com/example/videotecha/service/impl/TheaterServiceImpl.java]: It's been mentioned to others, so I have to mention it here as well: Do not use api related exceptions in the service layer. Best replace this with RuntimeException for now.

**Review by pantelicn — CHANGES_REQUESTED** [review]

**Review by pantelicn — APPROVED** [review]

File: src/main/java/com/example/videotecha/service/impl/TheaterServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    public List<Theater> findAll() {
+        return theaterRepository.findAll();
+    }
+
+    @Override
+    public Theater findById(Long id) {
+        return theaterRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("There is no theater with this id."));
+    }
+}
```
Comment (branko-paskali-valcon) [inline src/main/java/com/example/videotecha/service/impl/TheaterServiceImpl.java:33]: Empty line is missing

**Pull request merged**